File: problem/2215_Find the Difference of Two Arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[][]}
 */
var findDifference = function(nums1, nums2) {
    const set1 = new Set(nums1);
    const set2 = new Set(nums2);
    for(let i=0; i<nums1.length; i++){
        if(set2.has(nums1[i])){
            set1.delete(nums1[i])
            set2.delete(nums1[i])
        }
    }
    return [[...set1], [...set2]]
};
// TC: O(n) //GPT說 set.delete() 的時間複雜度是O(1) 
// MC: O(n)

var findDifference = function(nums1, nums2) {
    const s1 = new Set(nums1);
    const s2 = new Set(nums2);
    
    const a1 = [...s1].filter(x => !s2.has(x));
    const a2 = [...s2].filter(x => !s1.has(x));
    
    return [a1, a2];
};

// TC: O(n)
// MC: O(n)

File: c#-solution/2215. Find the Difference of Two Arrays.cs
// 2215. Find the Difference of Two Arrays
// https://leetcode.com/problems/find-the-difference-of-two-arrays/description/
// Tag: C#, Array, HashSet, LINQ, Easy

public class hashSetSolution {
    public IList<IList<int>> FindDifference(int[] nums1, int[] nums2) {
        var set1 = new HashSet<int>(nums1);
        var set2 = new HashSet<int>(nums2);

        return new List<IList<int>>{
            set1.Where(x => !set2.Contains(x)).ToList<int>(),
            set2.Where(x => !set1.Contains(x)).ToList<int>(),
        };
    }
}